src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -120,33 +134,36 @@ class spark_resource_adaptor final : public rmm::mr::device_memory_resource {
   std::map<long, std::set<long>> task_to_threads;
   JavaVM *jvm;
 
+  void throw_java_exception(const char * ex_class_name, const char* msg) {
+    JNIEnv *env = cudf::jni::get_jni_env(jvm);
+    //TODO: we may want to cache these in the future
+    jclass ex_class = env->FindClass(ex_class_name);
+    if (ex_class != nullptr) {
+      env->ThrowNew(ex_class, msg);
+    }
+    throw cudf::jni::jni_exception(msg);
```

Comment:
Nit: Now that this isn't trying to cache classes anymore, we can reuse the method from cudf::jni.
```suggestion
    cudf::jni::throw_java_exception(env, ex_class_name, msg);
```

Reply:
Done https://github.com/NVIDIA/spark-rapids-jni/pull/950/commits/ecc302daf5d2b4c3d33fe1277ce1c139e8d80a51